setup.py

```diff
@@ -13,6 +14,31 @@
 except ImportError:
     readme = desc
 
+
+def setup(*args, **kwargs):
+    """This wrapper exists to force the option --build-type=ReleaseWithDocs.
+
+    Neither Release nor RelWithDebInfo will work, due to hard-coded options in
+    scikit-build's UseCython.cmake that disable docstrings. The choice of
+    ReleaseWithDocs is arbitrary, as a string that won't overlap with any build
+    type handled in UseCython.cmake. See this issue for details:
+    https://github.com/scikit-build/scikit-build/issues/518
+    """
+    BUILD_TYPE = '--build-type=ReleaseWithDocs'
```

Comment:
I would prefer to use something that is clearly a fake value `--build-type=DummyType` or similar. Otherwise someone will inevitably see this at some point and think it's meaningful, no matter how well you document it. Giving it an apparently meaningful name is misleading.

Reply:
I disagree. The important part is that the name is unique and findable in the source. The name is arbitrary but that doesn't mean that it's a meaningless choice. I would rather convey the intent of the value in its name and avoid the possibility of raising false alarms when users see the build output and ask "Why is this saying I'm building a `FakeValue`? Seems wrong." (I would totally see `--build-type=FakeValue` and worry.)

Reply:
We have opposite reactions. I would see ReleaseWithDocs and spend a lot of time Googling it or searching CMake code to find what that means. As someone who is relatively new to CMake but has built enough projects to know that such names are common, wouldn't you interpret that name as something meaningful unless you wrote it?

I think we are maybe coming at it from different perspectives though. You are coming at this from the perspective of a user installing this, but actually inspecting the build output. In that case I see your point. I'm coming at this from the perspective of a developer who doesn't know CMake all that well. In that case looking at the CMake output or the CMakeCache.txt file would make it seem like there's some special feature being used. So maybe getting an opinion from @b-butler @tommy-waltmann and @klywang would be helpful.

Reply:
I think documenting the build type option is what's most important here. I like that the name of the build type option means something, but I think we do need to clarify that this option is our custom build type. Some comments in the CMake where the build option is defined would be helpful here, as well as in setup.py, and in the package documentation would make the intent of this build type clear to both developers and users who are building from source.

Reply:
I second @tommy-waltmann's comment that documentation is most important.

With that being said, I think a descriptive name like `ReleaseWithDocs` is helpful. If I understand correctly, running `python setup.py install` without the `--build-type=ReleaseWithDocs` flag or with `--build-type=Release` would result in a build without documentation. So, even if the name is arbitrary,  a label like `ReleaseWithDocs` makes the purpose of this additional option very clear, while an option like `DummyType` would raise questions of its usefulness and purpose.

Reply:
@tommy-waltmann @klywang Thanks for the feedback! I will make sure this is documented in the CMake help strings and package docs as well as `setup.py`. Also just to clarify, the point of this wrapper in `setup.py` is so that `python setup.py install` "just works" without needing to specify a `--build-type` option (we default to building with docs).

Reply:
OK seems like everyone is in agreement that the most important thing is lots of docs. 

I don't know CMake well enough to know if the extra variables you're setting are comprehensively overriding the behavior. However, since this is just a temporary patch until your scikit-build PR gets merged (and released) I'm OK with moving forward with this.

Reply:
> I don't know CMake well enough to know if the extra variables you're setting are comprehensively overriding the behavior.

I am decently confident that I'm defining a "comprehensive" set of variables here. I did a lot of reading on this and found several examples of custom configurations that match the behavior in this PR. I would say this is robust / not-hacky.